sark/tseries.py

```diff
@@ -0,0 +1,171 @@
+        means they are counted starting at 1 instead of 0.  Set this to False
+        if that is the case.
+
+    **kwards : Dict
+        Other keyword arguments passed on to the reader backend.  Any options
+        passed here takes precedence, and overwrites other values inferred from
+        the earlier keyword arguments.
+
+    Returns
+    -------
+    ts : Series/DataFrame
+        The time series is returned as a series or a dataframe depending on the
+        number of other columns that are present.
+
+    Examples
+    --------
+
+    To skip specific rows, maybe because they have bad data, or are empty, you
+    may use the `skiprows` option.  It can be set to a list-like where the
+    entries are row indices (numbers).
+
+    >>> read_timeseries("mydata.csv", source_t="table", col_units="hour",
+    ...     skiprows=range(1522, 5480))  # doctest: +SKIP
+
+    The above example skips rows 1522-5480.
+
+    Similarly, data type of the column values can be controlled by using the
+    `dtype` option.  When set to a `numpy.dtype`, all values will be read as
+    that type, which is probably relevant for the "table" mode.  In the
+    "multicol" mode, the types of the values can be controlled at the column
+    level by setting it to a dictionary, where the key matches a column name,
+    and the value is a valid `numpy.dtype`.
+
+    """
+    # FIXME: parse_dates & index_col assumes input is oriented as portrait
+    if source_t == "table":
+        if col_units is None:
+            raise ValueError("col_units: missing time unit for columns")
+        ts = from_table(
+            fpath, col_units=col_units, zero_idx=zero_idx, **kwargs,
+        )
+    elif source_t == "multicol":
+        if date_cols is None:
+            raise ValueError("date_cols: missing list of datetime columns")
+        ts = from_multicol(fpath, date_cols=date_cols, **kwargs,)
+    else:
+        if source_t:
+            warn(f"{source_t}: unsupported source, falling back to default")
+        ts = pd.read_csv(fpath, **kwargs)
+    return ts
+
+
+def from_table(fpath: _path_t, *, col_units: str, zero_idx: bool, **kwargs):
+    """Read a time series from a tabular file.
+
+    See Also
+    --------
+    read_timeseries : see for full documentation, main entrypoint for users
+
+    """
+    assert col_units in _time_units
+
+    # NOTE: assumption: input is oriented as portrait
+    ts = pd.read_csv(
+        fpath, **{"parse_dates": True, "index_col": 0, **kwargs}
+    ).stack()
+    # FIXME: do we also need support for custom stack level?  How common is a
+    # hierarchical index for columns
+
+    # merge indices
+    ts_idx = pd.DataFrame(ts.index.to_list())
+    ts_delta = pd.to_timedelta(
+        ts_idx.iloc[:, 1].astype(int) - int(not zero_idx), unit=col_units
+    )
+    ts.index = ts_idx.iloc[:, 0] + ts_delta
+    return ts
+
+
+def from_multicol(fpath: _path_t, *, date_cols: List[_col_t], **kwargs):
```

Comment:
Can this handle the time column preceding the date column?

Reply:
I believe you just need to order the list accordingly, so `date_cols=[0, 1]` vs `date_cols=[1, 0]`.  I should add a test for this.  Also, I don't know how this should be adapted when providing a custom datetime format, but that's probably best addressed in then.